src/main/java/com/auth/auth/controllers/SiteUserController.java

```diff
@@ -56,7 +56,36 @@ public String getSignUpPage() {
 
 
     @GetMapping("/myprofile")
-    public String getMyProfile() { return "my-profile.html"; }
+    public String getMyProfile(Model m, Principal p) {
+        SiteUser profileUser = null;
+        if (p != null) {
+            String username = p.getName();
+            profileUser = siteUserRepository.findByUsername(username);
+            m.addAttribute("username", profileUser.getUsername());
+        }
+
+        m.addAttribute("profileUsername", profileUser.getUsername());
+        m.addAttribute("firstName", profileUser.getFirstName());
+        m.addAttribute("lastName", profileUser.getLastName());
+        m.addAttribute("profileId", profileUser.getId());
+        m.addAttribute("profileDateCreated", profileUser.getDateCreated());
+
+        return "my-profile.html";
+    }
+
+    @GetMapping("/myprofile/following")
+    public String getFollowingPosts(Model m, Principal p){
+        SiteUser profileUser = null;
+        if (p != null) {
+            String username = p.getName();
+            profileUser = siteUserRepository.findByUsername(username);
+            m.addAttribute("username", profileUser.getUsername());
+            m.addAttribute("followingPosts", profileUser.getUsersIFollowPosts());
+        }
+
+
+        return "following-posts.html";
+    }
 
     @PostMapping("/signup")
     public RedirectView createUser(String username, String password) {
```

Comment:
When you create users here, you're only saving their username and password - the first and last name, bio, etc. are not being associated with the user before they're saved in the database. When you try to access that data, you'll get an error, because those fields are null.